src/Hl7.Fhir.ElementModel/Types/Date.cs

```diff
@@ -137,30 +137,39 @@ private static bool tryParse(string representation, out Date value)
             return success;
         }
 
-        public static Date operator +(Date me, Quantity value)
+        public static Date operator +(Date dateValue, Quantity addValue)
         {
-            var dto = me._parsedValue;
-            switch (value.Unit)
+            if (dateValue is null) throw new ArgumentNullException(nameof(dateValue));
+            if (addValue is null) throw new ArgumentNullException(nameof(addValue));
+
+            var dto = dateValue._parsedValue;
+            switch (addValue.Unit)
             {
                 // we can ignore precision, as the precision will "trim" it anyway, and if we add 13 months, then the year can tick over nicely
                 case "a": // UCUM
                 case "years":
                 case "year":
-                    dto = me._parsedValue.AddYears((int)value.Value);
+                    dto = dateValue._parsedValue.AddYears((int)addValue.Value);
```

Comment:
Wouldn't this crash when `addValue.Value is null`?

Reply:
this is the thing I mentioned above that I need to check and prevent in the Null Argument checking that I missed.

Reply:
addValue.value is a decimal, not an object or nullable decimal, so can't be null.